packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -1,34 +1,37 @@
 import React from 'react';
 import { AlignedData } from 'uplot';
-import { DataFrame, FieldMatcherID, fieldMatchers, TimeRange, TimeZone } from '@grafana/data';
+import { DashboardCursorSync, DataFrame, FieldMatcherID, fieldMatchers, TimeRange, TimeZone } from '@grafana/data';
 import { withTheme } from '../../themes';
 import { Themeable } from '../../types';
 import { UPlotConfigBuilder } from '../uPlot/config/UPlotConfigBuilder';
 import { GraphNGLegendEvent, XYFieldMatchers } from './types';
 import { preparePlotConfigBuilder, preparePlotFrame } from './utils';
-import { pluginLog, preparePlotData } from '../uPlot/utils';
+import { pluginLog, pluginLogger, preparePlotData } from '../uPlot/utils';
 import { PlotLegend } from '../uPlot/PlotLegend';
 import { UPlotChart } from '../uPlot/Plot';
+
 import { LegendDisplayMode, VizLegendOptions } from '../VizLegend/models.gen';
 import { VizLayout } from '../VizLayout/VizLayout';
+import { PlotSyncConfig } from '../uPlot/context';
 
 /**
  * @internal -- not a public API
  */
 export const FIXED_UNIT = '__fixed';
 
 export interface GraphNGProps extends Themeable {
+  data: DataFrame[];
   width: number;
   height: number;
-  data: DataFrame[];
-  structureRev?: number; // a number that will change when the data[] structure changes
   timeRange: TimeRange;
   legend: VizLegendOptions;
   timeZone: TimeZone;
+  structureRev?: number; // a number that will change when the data[] structure changes
   fields?: XYFieldMatchers; // default will assume timeseries data
   onLegendClick?: (event: GraphNGLegendEvent) => void;
   onSeriesColorChange?: (label: string, color: string) => void;
-  children?: (builder: UPlotConfigBuilder, alignedDataFrame: DataFrame) => React.ReactNode;
+  children?: (builder: UPlotConfigBuilder, alignedDataFrame: DataFrame, debug?: () => boolean) => React.ReactNode;
+  sync?: PlotSyncConfig | null;
```

Comment:
think we can use PanelContext & event bus for this.

Reply:
> very confused by the new context, thought we where to use PanelContext & the event bus, (which we will need to use these events in other non uPlot panels)

this is POC only for seeing how this feels with uPlot's built in feature

Reply:
the rest is on todos list above, I mean the panel's event bus events ^^ :)